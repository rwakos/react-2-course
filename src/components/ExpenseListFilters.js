import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import {setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate} from "../actions/filters";

export class ExpenseListFilters extends React.Component {
	state = {
		calendarFocused: null
	};

	onDatesChange = ({startDate, endDate}) => {
		this.props.setStartDate(startDate);
		this.props.setEndDate(endDate);
	};

	onFocusChange = (calendarFocused) => {
		this.setState(() => ({ calendarFocused }))
	};

	onTextChange = (e) => {
		this.props.setTextFilter(e.target.value);
	};

	onSortChange = (e)=>{
		if (e.target.value === 'date'){
			this.props.sortByDate();
		} else if (e.target.value === 'amount') {
			this.props.sortByAmount();
		}
	}
	render() {
		return (
			<div>
				<input type="text"
					   value={this.props.filters.text}
					   onChange={this.onTextChange} />

				<select
					value={this.props.filters.sortBy}
					onChange={this.onSortChange}
				>
					<option value="date">Date</option>
					<option value="amount">Amount</option>
				</select>

				<DateRangePicker
					startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
					endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
					onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
					focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
					onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
					numberOfMonths={1}
					isOutsideRange={() => false}
				/>
			</div>
		);
	}
};

const mapDispatchToProps = (dispatch) => ({
	setTextFilter: (text) => dispatch(setTextFilter(text)),
	sortByDate: () => dispatch(sortByDate()),
	sortByAmount:() => dispatch(sortByAmount()),
	setStartDate: (startDate) => dispatch(setStartDate(startDate)),
	setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

const mapStateToProps = (state) =>{
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
