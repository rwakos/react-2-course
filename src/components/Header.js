import React from 'react';
import {BrowserRouter, Route, Switch, Link, NavLink} from 'react-router-dom'; //
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}> Home</NavLink>
        <NavLink to="/create" activeClassName="is-active" exact={true}> Create</NavLink>
        <NavLink to="/help" activeClassName="is-active" exact={true}> Help</NavLink>
        <button onClick={startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);