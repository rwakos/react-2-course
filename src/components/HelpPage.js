import React from 'react';

const HelpPage = () => (
    <div>
        My Add HelpPage
    </div>
);

export default HelpPage;