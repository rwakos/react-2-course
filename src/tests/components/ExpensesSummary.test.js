import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render correct with 1 expense ExpensesSummary', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={235} />);
	expect(wrapper).toMatchSnapshot();
});

test('should render correct with multiple expenses ExpensesSummary', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={23} expenseTotal={231321321} />);
	expect(wrapper).toMatchSnapshot();
});