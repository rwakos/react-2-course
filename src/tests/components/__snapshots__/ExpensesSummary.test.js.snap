// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct with 1 expense ExpensesSummary 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling: 
    $2.35
  </h1>
</div>
`;

exports[`should render correct with multiple expenses ExpensesSummary 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling: 
    $2,313,213.21
  </h1>
</div>
`;
